### PR TITLE
Make open writers limit configurable in iceberg connector

### PR DESCRIPTION
**Pull request by v-jizhang**
Cherry-pick of Trino https://github.com/trinodb/trino/pull/6684

Addresses https://github.com/trinodb/trino/issues/6650.

Co-authored-by: Akshay Rai <akrai@linkedin.com>

```
== RELEASE NOTES ==

Iceberg Changes
* Add `iceberg.max-partitions-per-writer` configuration property to allow configuring the limit on partitions per writer. 

```

### CODE REVIEW

**Review by NikhilCollooru — APPROVED** [review]

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSinkProvider.java [hunk 1]
```diff
@@ -43,18 +43,22 @@
...
+    private final int maxOpenPartitions;
 
     @Inject
     public IcebergPageSinkProvider(
             HdfsEnvironment hdfsEnvironment,
             JsonCodec<CommitTaskData> jsonCodec,
             IcebergFileWriterFactory fileWriterFactory,
-            PageIndexerFactory pageIndexerFactory)
+            PageIndexerFactory pageIndexerFactory,
+            IcebergConfig config)
```
Comment (kewang1024) [inline presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSinkProvider.java:54]: NIT: icebergConfig

**Review by kewang1024 — APPROVED** [review]

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 2]
```diff
@@ -45,6 +45,9 @@ Property Name                          Description
 ``iceberg.catalog.cached-catalog-num`` The number of Iceberg catalogs to cache.
 
 ``iceberg.hadoop.config.resources``    The path(s) for Hadoop configuration resources.
+
```
Comment (kewang1024) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:48]: Can we also cherry pick and improve the rest of the property documentation?
For example, for `iceberg.file-format`, can we also add something like "Available values are``PARQUET``, ``ORC``, defaults to ``ORC ``". Same for the rest of the configs

**Review by kewang1024 — COMMENTED** [review]
For the second commit, can we improve the commit message to "Add Iceberg configuration properties in docs"

File: presto-docs/src/main/sphinx/connector/iceberg.rst (hunk 2 above)
Comment (v-jizhang) [reply presto-docs/src/main/sphinx/connector/iceberg.rst:48]: > Can we also cherry pick and improve the rest of the property documentation? For example, for `iceberg.file-format`, can we also add something like "Available values are`PARQUET`, `ORC`, defaults to `ORC `". Same for the rest of the configs

Presto already has that information.

Comment (v-jizhang) [reply presto-docs/src/main/sphinx/connector/iceberg.rst:48]: It is in the doc, please see line #63

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSinkProvider.java [hunk 3]
```diff
@@ -51,14 +51,14 @@ public IcebergPageSinkProvider(
...
+            IcebergConfig icebergConfig)
     {
         this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
         this.jsonCodec = requireNonNull(jsonCodec, "jsonCodec is null");
         this.fileWriterFactory = requireNonNull(fileWriterFactory, "fileWriterFactory is null");
         this.pageIndexerFactory = requireNonNull(pageIndexerFactory, "pageIndexerFactory is null");
-        requireNonNull(config, "config is null");
-        this.maxOpenPartitions = config.getMaxPartitionsPerWriter();
+        requireNonNull(icebergConfig, "icebergConfig is null");
+        this.maxOpenPartitions = icebergConfig.getMaxPartitionsPerWriter();
```
Comment (NikhilCollooru) [inline presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSinkProvider.java:61]: This change should be in the first commit.

**Review by NikhilCollooru — COMMENTED** [review]
Please change the 2nd commit title to `"Add Iceberg configuration property to docs"`

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSinkProvider.java (hunk 3 above)
Comment (v-jizhang) [reply presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSinkProvider.java:61]: Moved. Thanks.

**Comment by v-jizhang** [comment]
> For the second commit, can we improve the commit message to "Add Iceberg configuration properties in docs"

Changed. Thank you.

**Pull request merged**